packages/source-map-utils/index.js

```diff
@@ -493,30 +493,32 @@ var SourceMapUtils = {
       const jumpIndex = instructions.findIndex(
         instruction => instruction.pc === jumpAddress
       );
-      if (checkAgainstTemplate(instructions, jumpIndex, newSequence)) {
-        return true;
-      }
-      debug("indirect: %O", instructions.slice(index, index + 4));
-      debug("jumpAddress: %d", jumpAddress);
-      debug("jumpIndex: %d", jumpIndex);
-      debug("instr count: %d", instructions.length);
-      const jumpInstruction = instructions[jumpIndex];
-      const jumpFile = jumpInstruction.file;
-      if (jumpFile !== -1) {
-        const findOverlappingRange = overlapFunctions[jumpFile];
-        const range = SourceMapUtils.getSourceRange(jumpInstruction);
-        const { node: jumpNode } = SourceMapUtils.findRange(
-          findOverlappingRange,
-          range.start,
-          range.length
-        );
-        if (
-          jumpNode &&
-          jumpNode.nodeType === "YulFunctionDefinition" &&
-          jumpNode.name === "panic_error_0x51"
-        ) {
+      if (jumpIndex !== undefined) {
```

Comment:
What's the return type if `jumpIndex` is `undefined`? If I'm reading it correctly, I think there's an implicit `undefined` return, while it returns `bool` if `jumpIndex` _is_ defined?

What are the implications to the caller if this function returns `undefined`?

Also I assume that you verified that `findIndex` strictly returns `undefined` when the search predicate is never satisfied? I know some library authors like to follow the pattern of `string.indexOf` and return `-1`.

Reply:
> What's the return type if jumpIndex is undefined? If I'm reading it correctly, I think there's an implicit undefined return, while it returns bool if jumpIndex is defined?

No, in this case, as I mentioned, it'll fall down and hit the `return false;` at the end.

> Also I assume that you verified that findIndex strictly returns undefined when the search predicate is never satisfied? I know some library authors like to follow the pattern of string.indexOf and return -1.


Oops!  You're absolutely right, it returns `-1`.  My mistake.  The problem with writing things you didn't test! :grimacing: Thanks for catching that, will go change that.

Reply:
To be clear, I think unless it's part of the design of this function to return `undefined` sometimes, it's likely better that it always return `bool`.

Reply:
I think we were commenting at the same time. Honestly I'm having some serious trouble even seeing where this function starts and where it ends given all of the nested control flow structure!

Reply:
Also my bad - I thought the end of the diff was the end of the file. I'm still having a "fun" time spotting the beginning of the function, however.

Reply:
ok, found it with confidence at 399. I'm apparently a bit cross-eyed today, but also yeah - cyclomatic complexity like woah.